client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx

```diff
@@ -0,0 +1,279 @@
+import React, { ComponentType, useState } from 'react';
+import {
+  rankWith,
+  uiTypeIs,
+  ArrayControlProps,
+  ControlProps,
+} from '@jsonforms/core';
+import {
+  withJsonFormsArrayControlProps,
+  withJsonFormsControlProps,
+} from '@jsonforms/react';
+import { Typography } from '@mui/material';
+import {
+  useTranslation,
+  useFormatDateTime,
+  BasicTextInput,
+  FlatButton,
+  EditIcon,
+} from '@openmsupply-client/common';
+import { z } from 'zod';
+import { useZodOptionsValidation } from '../../hooks/useZodOptionsValidation';
+
+import { DateUtils } from '@common/intl';
+import { JsonData } from '../../JsonForm';
+import {
+  CommonOptions,
+  ArrayControlCustomProps,
+  ArrayCommonComponent,
+} from './';
+import { useDebouncedTextInput } from '../../hooks/useDebouncedTextInput';
+
+export const notesTester = rankWith(6, uiTypeIs('Notes'));
+
+const NotesOptions = CommonOptions.extend({
+  editRestrictions: z
+    .object({
+      /**
+       * If true, only the newest note can be edited
+       */
+      latest: z.boolean().optional(),
+      /**
+       * If true, the element's data.authorId field must match the currently logged-in user (used for Notes)
+       */
+      isCurrentUser: z.boolean().optional(),
+      /**
+       * Number in days. Timestamp (data.created) must be less than this many
+       * days in the past (used for Notes)
+       */
+      maxAge: z.number().optional(),
+      // Add more as required
+    })
+    .optional(),
+});
+
+const NotesComponent = (props: ArrayControlCustomProps) => {
+  const { enabled, data, config } = props;
+  const { localisedDateTime } = useFormatDateTime();
+
+  const options = props.uischema.options;
+
+  const inputData = data ?? [];
+
+  // This injects the required details info so that the children of "Notes" will
+  // be "Note" components without having to specify it specifically in the
+  // UISchema (since that would be redundant)
+  if (!props.uischema.options) props.uischema.options = {};
+  if (!props.uischema?.options?.['detail']) {
+    props.uischema.options['detail'] = {
+      type: 'VerticalLayout',
+      elements: [
+        {
+          type: 'Note',
+          scope: '/properties',
+        },
+      ],
+    };
+  }
```

Comment:
This is a little bit hacky, but I could really find a more elegant way to get the Array to know what it's children should be if not specified in the UI schema.

Reply:
You could probably pass in the child renderer somehow but I think this is actually easier to understand as it is. As you mentioned before as well, we are probably on the verge where it might become simpler to just copy the core array code if the component diverge even further...